nodes/src/nodes/tool_shell/IInstance.py

```
@@ -0,0 +1,160 @@
+            command,
+            cwd=cwd,
+            env=env,
+            timeout=timeout,
+            max_output_bytes=self.IGlobal.max_output_bytes,
+        )
+
+    def _validate_command(self, command: str) -> None:
+        """Reject commands that don't match any configured allowlist regex."""
+        patterns = self.IGlobal.command_patterns or []
+        if patterns and not any(p.search(command) for p in patterns):
+            raise ValueError('Command is not permitted by the configured allowlist.')
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)